content/developer/reference/user_interface/view_architectures.rst

```diff
@@ -0,0 +1,3710 @@
+   number of records is displayed instead.
+
+   :requirement: Optional
+   :type: str
+   :default: `''`
+
+.. admonition:: Possible structure and representation of its rendering
+
+   .. list-table::
+      :class: o-showcase-table
+
+      * - .. image:: view_architectures/kanban_progressbar.svg
+             :align: center
+
+      * - .. code-block:: xml
+
+             <kanban>
+                 <progressbar field="activity_state"
+                              colors="{'planned': 'success', 'today': 'warning', 'overdue': 'danger'}"
+                              sum_field="expected_revenue"/>
+                 <templates>
+                     ...
+                 </templates>
+             </kanban>
+
+.. _reference/view_architectures/kanban/templates:
+
+`templates`: define cards structure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The `templates` elements is used to define the :ref:`QWeb templates <reference/qweb>` that structure
+the kanban cards.
+
+Cards structure definition can be split into multiple templates for clarity, but at least one root
+template `kanban-box` must be defined.
+
+Two additional templates can be defined: `kanban-menu` and `kanban-tooltip`. If defined, the
+`kanban-menu` template is rendered inside a dropdown that can be toggled with a vertical ellipsis
+(:guilabel:`⋮`) on the top right of the card. The `kanban-tooltip` template is rendered inside a
+tooltip when hovering kanban cards.
+
+The templates are written in :ref:`JavaScript QWeb <reference/qweb/javascript>`
+
+.. code-block:: xml
+
+   <kanban>
+       ...
+       <templates>
+           <t t-name="kanban-box">
+               <div>
+                   <field name="name"/>
+               </div>
+           </t>
+       </templates>
+   </kanban>
+
+The following variables are available in the rendering context:
+
+.. attribute:: widget
+   :noindex:
+
+   The current :js:class:`KanbanRecord`. It can be used to fetch some meta-information. The methods
+   are also available directly in the template context and don't need to be accessed via `widget`.
+
+   :type: str
+
+.. attribute:: record
+   :noindex:
+
+   An object with all the requested fields as its attributes. Each field has two attributes: `value`
+   and `raw_value`. The former is formatted according to current user parameters while the latter is
+   the raw value from a :meth:`~odoo.models.Model.read` (except for the `date` and `datetime` fields
+   that are `formatted according to the user locale <https://github.com/odoo/odoo/blob/a678bd4e
+   /addons/web_kanban/static/src/js/kanban_record.js#L102>`_).
+
+   :type: str
+
+.. attribute:: context
+   :noindex:
+
+   The current context propagated from either the action that opens the kanban view, or the one2many
+   or many2many field that embeds the kanban view in a form view.
+
+   :type: str
+
+.. attribute:: user_context
+   :noindex:
+
+   :type: str
```

Comment:
I'd avoid mentioning that one, it's not used anywhere (and maybe included in the standard context)